4_Zadania_dodatkowe/6.InfiniteSlider/js/app.js

```diff
@@ -0,0 +1,71 @@
+$(document).ready(function () {
+   // console.log('test');
+
+    var elementPrevButton = $('#prevPicture');
+    var elementNextButton = $('#nextPicture');
+    // console.log(elementPrevButton, elementNextButton);
+
+    var elementListLi = $('li');
+    // console.log(elementListLi);
+
+    var elementUl = $('ul');
+    // console.log(elementUl);
+
+    var visibleImageIndex = 1;
+    var imageWidth = 404;
```

Comment:
your script should not assume image's width
you can read it from matched element using http://api.jquery.com/outerWidth/
this is because your script should work well with images of different size without modification

Reply:
I tried outerwidth(), css('width') combined with reset as well but nonetheless I still get 400px instead of 404px so the effect is not satisfactory

Reply:
I think these extra 4 pixels are caused by a whitespace between `<li>` elements.
See https://css-tricks.com/fighting-the-space-between-inline-block-elements/ for explanation and possible solutions. I'd say you may want to use `font-size` technique in this case.